smarts/env/gymnasium/hiway_env_v1.py

```diff
@@ -0,0 +1,390 @@
+            dones["__all__"],
+            infos,
+        )
+
+    def reset(
+        self, *, seed: Optional[int] = None, options: Optional[Dict[str, Any]] = None
+    ) -> Tuple[ObsType, Dict[str, Any]]:
+        """Resets the environment to an initial internal state, returning an initial observation and info.
+        This method generates a new starting state often with some randomness to ensure that the agent explores the
+        state space and learns a generalised policy about the environment. This randomness can be controlled
+        with the ``seed`` parameter otherwise if the environment already has a random number generator and
+        :meth:`reset` is called with ``seed=None``, the RNG is not reset.
+        Therefore, :meth:`reset` should (in the typical use case) be called with a seed right after initialization and then never again.
+        For Custom environments, the first line of :meth:`reset` should be ``super().reset(seed=seed)`` which implements
+        the seeding correctly.
+        .. versionchanged:: v0.25
+            The ``return_info`` parameter was removed and now info is expected to be returned.
+        Args:
+            seed (optional int): The seed that is used to initialize the environment's PRNG (`np_random`).
+                If the environment does not already have a PRNG and ``seed=None`` (the default option) is passed,
+                a seed will be chosen from some source of entropy (e.g. timestamp or /dev/urandom).
+                However, if the environment already has a PRNG and ``seed=None`` is passed, the PRNG will *not* be reset.
+                If you pass an integer, the PRNG will be reset even if it already exists.
+                Usually, you want to pass an integer *right after the environment has been initialized and then never again*.
+                Please refer to the minimal example above to see this paradigm in action.
+            options (optional dict): Additional information to specify how the environment is reset (optional,
+                depending on the specific environment)
+        Returns:
+            observation (ObsType): Observation of the initial state. This will be an element of :attr:`observation_space`
+                (typically a numpy array) and is analogous to the observation returned by :meth:`step`.
+            info (dictionary):  This dictionary contains auxiliary information complementing ``observation``. It should be analogous to
+                the ``info`` returned by :meth:`step`.
+        """
+        super().reset(seed=seed, options=options)
+        scenario = next(self._scenarios_iterator)
+
+        self._dones_registered = 0
+        observations = self._smarts.reset(scenario)
+        info = {}
+
+        if self._env_renderer is not None:
+            self._env_renderer.reset(observations)
+
+        if seed is not None:
+            smarts_seed(seed)
+        return self.observations_formatter.format(observations), info
```

Comment:
The (i) contents (with all options enabled), (ii) types, and (iii) structure, of the observation returned by the env (i.e. `self.observations_formatter.format(observations)`) should desirably be defined and documented in a manner easy to refer for users.

Reply:
That was part of the reason for applying it to the action space. In that case the users should just be able to call `action_space` to see the format of the data.

I will add documentation on Thursday/Friday to describe the observations in full.